lock-api/src/test/java/com/palantir/lock/client/LockRefresherTest.java

```diff
@@ -95,6 +107,51 @@ public void doesNotFailIfDelegateThrows() {
         verify(timelock, times(2)).refreshLockLeases(lockTokensToRefresh);
     }
 
+    @Test
+    public void locksWithoutSpecifiedOptionsHaveLongTenure() {
+        when(timelock.refreshLockLeases(TOKENS)).thenReturn(TOKENS);
+        when(clock.instant()).thenReturn(Instant.EPOCH);
+        refresher.registerLocks(TOKENS);
+
+        tick();
+        verify(timelock).refreshLockLeases(TOKENS);
+
+        when(clock.instant()).thenReturn(Instant.MAX); // O(1 billion years). AtlasDB probably won't exist then...?
+        tick();
+        verify(timelock, times(2)).refreshLockLeases(TOKENS);
+        tick();
+        verify(timelock, times(3)).refreshLockLeases(TOKENS);
+    }
+
+    @Test
+    public void lockNoLongerRefreshedAfterTenureElapses() {
+        Runnable callback = mock(Runnable.class);
+
+        when(timelock.refreshLockLeases(TOKENS)).thenReturn(TOKENS);
+        when(clock.instant()).thenReturn(Instant.EPOCH);
+        refresher.registerLocks(
+                TOKENS,
+                ClientLockingOptions.builder()
+                        .maximumLockTenure(Duration.ofMinutes(5))
+                        .tenureExpirationCallback(callback)
+                        .build());
+
+        tick();
+        verify(timelock).refreshLockLeases(TOKENS);
+        verify(callback, never()).run();
+
+        when(clock.instant()).thenReturn(Instant.EPOCH.plus(Duration.ofMinutes(3)));
+        tick();
+        verify(timelock, times(2)).refreshLockLeases(TOKENS);
+        verify(callback, never()).run();
+
+        when(clock.instant()).thenReturn(Instant.EPOCH.plus(Duration.ofMinutes(7)));
+        tick();
+        verify(timelock, times(2)).refreshLockLeases(TOKENS);
+        // Callback ran once per expired token
```

Comment:
tick once more to make sure callbacks have run exactly once?

Reply:
... and looks like you caught a bug here! 🚀